crates/bevy_render/src/mesh/primitives/dim3/torus.rs

```diff
@@ -0,0 +1,167 @@
+use bevy_math::{primitives::Torus, Vec3};
+use wgpu::PrimitiveTopology;
+
+use crate::{
+    mesh::{Indices, Mesh, Meshable},
+    render_asset::RenderAssetUsages,
+};
+
+/// A builder used for creating a [`Mesh`] with a [`Torus`] shape.
+#[derive(Clone, Copy, Debug)]
+pub struct TorusMeshBuilder {
+    /// The [`Torus`] shape.
+    pub torus: Torus,
+    /// The number of vertices used for each circular segment
+    /// in the ring or tube of the torus.
+    ///
+    /// The default is `24`.
+    pub minor_resolution: usize,
+    /// The number of segments used for the main ring of the torus.
+    ///
+    /// A resolution of `4` would make the torus appear rectangular,
+    /// while a resolution of `32` resembles a circular ring.
+    ///
+    /// The default is `32`.
+    pub major_resolution: usize,
+}
+
+impl Default for TorusMeshBuilder {
+    fn default() -> Self {
+        Self {
+            torus: Torus::default(),
+            minor_resolution: 24,
+            major_resolution: 32,
+        }
+    }
+}
+
+impl TorusMeshBuilder {
+    /// Creates a new [`TorusMeshBuilder`] from an inner and outer radius.
+    ///
+    /// The inner radius is the radius of the hole, and the outer radius
+    /// is the radius of the entire object.
+    #[inline]
+    pub fn new(inner_radius: f32, outer_radius: f32) -> Self {
+        Self {
+            torus: Torus::new(inner_radius, outer_radius),
+            ..Default::default()
+        }
+    }
+
+    /// Sets the number of vertices used for each circular segment
+    /// in the ring or tube of the torus.
+    #[inline]
+    pub const fn minor_resolution(mut self, resolution: usize) -> Self {
+        self.minor_resolution = resolution;
+        self
+    }
+
+    /// Sets the number of segments used for the main ring of the torus.
+    ///
+    /// A resolution of `4` would make the torus appear rectangular,
+    /// while a resolution of `32` resembles a circular ring.
+    #[inline]
+    pub const fn major_resolution(mut self, resolution: usize) -> Self {
+        self.major_resolution = resolution;
+        self
+    }
+
+    /// Builds a [`Mesh`] according to the configuration in `self`.
+    pub fn build(&self) -> Mesh {
+        // code adapted from http://apparat-engine.blogspot.com/2013/04/procedural-meshes-torus.html
+        // (source code at https://github.com/SEilers/Apparat)
```

Comment:
This link doesn't work

Reply:
Not my link, copied from the original torus implementation. Looks like the repo has probably been deleted though, so I can remove this link unless I find it

Reply:
```suggestion
```